src/index.js

```diff
@@ -51,20 +51,24 @@ const authenticate = async (db) => {
   headers.set('Content-Type', 'application/json');
   headers.set('Accept', 'application/json');
 
-  const body = JSON.stringify({ name: db.credentials.username, password: db.credentials.password});
+  const body = JSON.stringify({ name: db.credentials?.username, password: db.credentials?.password});
```

Comment:
What happens if this request goes through with undefined name and password? We've wiped the basic auth from the URL so isn't this guaranteed to fail?

Reply:
Well, we take auth from either the URL or the passed `auth` property and save them in `credentials`. If they existed in either place, then we pass them to this request, so it won't fail. 

The session request doesn't need basic auth to succeed, passing values as post payload is enough.

Reply:
oh, true, we can do that.